File: aspnet-core/src/Project_ABP.Application/Services/HospitalService.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Microsoft.Extensions.Logging;
using Project_ABP.Dto.HospitalDtos;
using Project_ABP.Entities;
using Project_ABP.Filter;
using Project_ABP.IRepositories;
using Project_ABP.IServices;
using Project_ABP.Permissions;
using Volo.Abp.Application.Dtos;
using Volo.Abp.Application.Services;
using Volo.Abp.Domain.Repositories;

namespace Project_ABP.Services
{
    public class HospitalService : CrudAppService<Hospital, HospitalDto, int, HospitalPagedAndSortedResultRequestDto, CreateOrUpdateHospitalDto, CreateOrUpdateHospitalDto>, IHospitalService
    {
        private readonly IHospitalRepository _hospitalRepository;
        private ILogger<HospitalService> _logger;
        public HospitalService(IRepository<Hospital, int> repository, IHospitalRepository hospitalRepository, ILogger<HospitalService> logger) : base(repository)
        {
            CreatePolicyName = Project_ABPPermissions.HospitalPermissions.Create;
            UpdatePolicyName = Project_ABPPermissions.HospitalPermissions.Edit;
            DeletePolicyName = Project_ABPPermissions.HospitalPermissions.Delete;

            _hospitalRepository = hospitalRepository;
            _logger = logger;
        }

        public async Task<List<HospitalDto>> GetAllHospitals(int maTinh, int maHuyen, int maXa)
        {
            try
            {
                var responses = await _hospitalRepository.GetAllHospitals(maTinh, maHuyen, maXa);
                return ObjectMapper.Map<List<Hospital>, List<HospitalDto>>(responses);
            }
            catch (Exception ex)
            {
                _logger.LogError(ex.Message);
                throw;
            }
        }

        public override async Task<PagedResultDto<HospitalDto>> GetListAsync(HospitalPagedAndSortedResultRequestDto request)
        {
            try
            {
                var filter = ObjectMapper.Map<HospitalPagedAndSortedResultRequestDto, HospitalFilter>(request);
                var sorting = string.IsNullOrEmpty(request.Sorting) ? "ma ASC" : request.Sorting;
                var response = await _hospitalRepository.GetListAsync(request.SkipCount, request.MaxResultCount, sorting, filter);
                var totalCount = await _hospitalRepository.GetTotalCountAsync(filter);
                return new PagedResultDto<HospitalDto>(totalCount, ObjectMapper.Map<List<Hospital>, List<HospitalDto>>(response));
            }
            catch (Exception ex)
            {
                _logger.LogError(ex.Message);
                throw;
            }
        }
    }
}

File: angular/src/app/proxy/services/hospital.service.ts
import { RestService, Rest } from '@abp/ng.core';
import type { PagedResultDto } from '@abp/ng.core';
import { Injectable } from '@angular/core';
import type { CreateOrUpdateHospitalDto, HospitalDto, HospitalPagedAndSortedResultRequestDto } from '../dto/hospital-dtos/models';

@Injectable({
  providedIn: 'root',
})
export class HospitalService {
  apiName = 'Default';
  

  create = (input: CreateOrUpdateHospitalDto, config?: Partial<Rest.Config>) =>
    this.restService.request<any, HospitalDto>({
      method: 'POST',
      url: '/api/app/hospital',
      body: input,
    },
    { apiName: this.apiName,...config });
  

  delete = (id: number, config?: Partial<Rest.Config>) =>
    this.restService.request<any, void>({
      method: 'DELETE',
      url: `/api/app/hospital/${id}`,
    },
    { apiName: this.apiName,...config });
  

  get = (id: number, config?: Partial<Rest.Config>) =>
    this.restService.request<any, HospitalDto>({
      method: 'GET',
      url: `/api/app/hospital/${id}`,
    },
    { apiName: this.apiName,...config });
  

  getAllHospitalsByMaTinhAndMaHuyenAndMaXa = (maTinh: number, maHuyen: number, maXa: number, config?: Partial<Rest.Config>) =>
    this.restService.request<any, HospitalDto[]>({
      method: 'GET',
      url: '/api/app/hospital/hospitals',
      params: { maTinh, maHuyen, maXa },
    },
    { apiName: this.apiName,...config });
  

  getList = (request: HospitalPagedAndSortedResultRequestDto, config?: Partial<Rest.Config>) =>
    this.restService.request<any, PagedResultDto<HospitalDto>>({
      method: 'GET',
      url: '/api/app/hospital',
      params: { maTinh: request.maTinh, maHuyen: request.maHuyen, maXa: request.maXa, filter: request.filter, sorting: request.sorting, skipCount: request.skipCount, maxResultCount: request.maxResultCount },
    },
    { apiName: this.apiName,...config });
  

  update = (id: number, input: CreateOrUpdateHospitalDto, config?: Partial<Rest.Config>) =>
    this.restService.request<any, HospitalDto>({
      method: 'PUT',
      url: `/api/app/hospital/${id}`,
      body: input,
    },
    { apiName: this.apiName,...config });

  constructor(private restService: RestService) {}
}
